Support Renesas RA SCI I2C driver

duynguyenxa opened the pull request:
This PR add support for SCI I2C driver on RA devices

dsseng commented on boards/renesas/ek_ra6m4/ek_ra6m4-pinctrl.dtsi:
```diff
@@ -12,6 +12,15 @@
+	sci2_default: sci2_default {
+		group1 {
+			/* sda scl */
+			psels = <RA_PSEL(RA_PSEL_SCI_2, 1, 12)>,<RA_PSEL(RA_PSEL_SCI_2, 1, 13)>;
```
I think this should follow the file style

```suggestion
			psels = <RA_PSEL(RA_PSEL_SCI_2, 1, 12)>,
				<RA_PSEL(RA_PSEL_SCI_2, 1, 13)>;
```

dsseng commented on boards/renesas/ek_ra6m5/ek_ra6m5-pinctrl.dtsi:
```diff
@@ -12,6 +12,15 @@
+	sci2_default: sci2_default {
+		group1 {
+			/* sda scl */
+			psels = <RA_PSEL(RA_PSEL_SCI_2, 1, 12)>,<RA_PSEL(RA_PSEL_SCI_2, 1, 13)>;
```
```suggestion
			psels = <RA_PSEL(RA_PSEL_SCI_2, 1, 12)>,
				<RA_PSEL(RA_PSEL_SCI_2, 1, 13)>;
```

dsseng reviewed (commented):
nit

khaicaorvc committed:
drivers: i2c: Initial support for i2c sci driver on Renesas RA

First commit to add support for Renesas RA i2c sci driver

Signed-off-by: Khai Cao <khai.cao.xh@renesas.com>

khaicaorvc committed:
dts: arm: add SCI I2C device node for RA6 series

Add I2C device nodes for the SCI peripheral on RA6 series

Signed-off-by: Khai Cao <khai.cao.xh@renesas.com>

khaicaorvc committed:
dts: arm: add SCI I2C device node for RA4 series

Add I2C device nodes for the SCI peripheral on RA4 series

Signed-off-by: Khai Cao <khai.cao.xh@renesas.com>

khaicaorvc committed:
dts: arm: add SCI I2C device node for RA2 series

Add I2C device nodes for the SCI peripheral on RA2 series

Signed-off-by: Khai Cao <khai.cao.xh@renesas.com>

khaicaorvc committed:
tests: i2c: add support for i2c sci on Renesas RA boards

Add board support for EK_RA6M5, EK_RA6M4 i2c test use sci_i2c:
- tests/drivers/i2c/i2c_api

Example for i2c use sci_i2c:

west build -b ek_ra6m5 tests/drivers/i2c/i2c_api/ -p always \
 -DDTC_OVERLAY_FILE=boards/ek_ra6m5_sci_i2c.overlay \
 -DCONF_FILE="boards/ek_ra6m5_sci_i2c.conf"

Signed-off-by: Khai Cao <khai.cao.xh@renesas.com>

thenguyenyf commented on drivers/i2c/i2c_renesas_ra_sci.c:
```diff
@@ -0,0 +1,731 @@
+	k_sem_give(&data->bus_lock);
+
+	/* Callback may wish to start another transfer */
+	cb(dev, result, p_context);
```
Better to have a NULL check
```suggestion
    if (cb) {
        cb(dev, result, p_context);
    }
```

thenguyenyf commented on drivers/i2c/i2c_renesas_ra_sci.c:
```diff
@@ -0,0 +1,731 @@
+	struct sci_i2c_data *data = dev->data;
+	fsp_err_t fsp_err;
+	struct i2c_msg *current, *next;
+
```
The blank line should be removed

thenguyenyf commented on drivers/i2c/i2c_renesas_ra_sci.c:
```diff
@@ -0,0 +1,731 @@
+					    void *p_context)
+{
+	struct sci_i2c_data *data = dev->data;
+
```
This blank line should be removed

thenguyenyf commented on drivers/i2c/i2c_renesas_ra_sci.c:
```diff
@@ -0,0 +1,731 @@
+	fsp_err_t fsp_err;
+	int ret;
+
+	data->dev_config |= I2C_MODE_CONTROLLER;
```
Shoud it be `=` instead of `|=`?

thenguyenyf commented on dts/bindings/i2c/renesas,ra-i2c-sci.yaml:
```diff
@@ -0,0 +1,34 @@
+
+include: [i2c-controller.yaml, pinctrl-device.yaml]
+
+properties:
+
```
```suggestion
properties:
```

thenguyenyf commented on drivers/i2c/i2c_renesas_ra_sci.c:
You need to re-run clang-format for this file

thenguyenyf commented:
You will need to rebase to pass this job.
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18088368313/job/51463326829?pr=96558

dsseng commented on drivers/i2c/i2c_renesas_ra_sci.c:
```diff
@@ -0,0 +1,739 @@
+/* FSP interruption handlers. */
+void sci_i2c_txi_isr(void);
+void sci_i2c_tei_isr(void);
+void sci_i2c_rxi_isr(void);
```
Doesn't FSP expose it anyhow? Could this be fixed there maybe?

What's the supposed way of using these, should FSP itself configure the IRQs, setting up these handlers?

teburd commented on drivers/i2c/i2c_renesas_ra_sci.c:
```diff
@@ -0,0 +1,744 @@
+		num_msgs--;
+	}
+
+RELEASE_BUS:
```
out: is a more common label for this in Zephyr (lower cased). Recommend following general Zephyr styling here.

duynguyenxa committed:
boards: renesas: Add SCI I2C node on Renesas RA boards

Enable support of i2c sci driver on these boards: ek_ra6m5,
ek_ra6m4

Signed-off-by: Khai Cao <khai.cao.xh@renesas.com>
Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>
